compiler-core/src/error.rs

```diff
@@ -1923,6 +1923,26 @@ These values are not matched:
                         }),
                     }
                 }
+
+                TypeError::IncorrectConstVarValueConstructor { location, name } => {
+                    let text = format!("In order to be used in a module constant definition, the name '{}' must refer to either a module constant or a module function.", name);
+                    Diagnostic {
+                        title: "Incorrect value constructor used in module constant definition"
```

Comment:
Hm...how about `Expected a function or a module constant`?

I thought I mentioned this somewhere but I may have forgotten -- I'm 99% sure it's impossible for the user to ever cause this error, so maybe it could be removed and replaced with an `unreachable` statement (unless that's bad style, I'm new to rust). The reasons for why it is impossible to cause are a little convoluted, so maybe we should just keep it in case something changes somewhere.

Reply:
Ah nice! Let's use `unreachable` with a description of why we think that is the case.